src/util/strconv.h

```diff
@@ -106,11 +110,97 @@ static char *rm_strndup_unescape(const char *s, size_t len) {
   return ret;
 }
 
+// transform utf8 string to lowercase using nunicode library
+// encoded: the utf8 string to transform
+// in_len: the length of the utf8 string
+// dst: the destination buffer to store the transformed string
+// returns the bytes written to encoded, or 0 if the length of the transformed
+// string is greater than in_len and no transformation was done
+static size_t nunicode_tolower(const char *encoded, size_t in_len, char *dst) {
+  char encoded_stack_buffer[SSO_MAX_LENGTH];
+  const char *encoded_copy = encoded_stack_buffer;
+  uint32_t u_stack_buffer[SSO_MAX_LENGTH * 2];
+  uint32_t *u_buffer = u_stack_buffer;
+
+  if (in_len == 0) {
+    return 0;
+  }
+
+  // create a zero-terminated copy of the encoded string
+  if (in_len >= SSO_MAX_LENGTH - 1) {
+    encoded_copy = rm_strndup(encoded, in_len);
```

Comment:
Can we avoid duplicating the string pointed by `encoded` and work on it directly? (since we are always using it as destination as well and we are already "destructive" and its size can only decrease from unescaping)

Reply:
Yes. Done in [b71f865](https://github.com/RediSearch/RediSearch/pull/5391/commits/b71f8656fa443d46a814923f21c05adf009a0ecb)